### DISCUSSION TITLE
How to activate ProgressRing before synchronous call

### DISCUSSION THREAD

**Discussion by greatoceansoftware** [microsoft/microsoft-ui-xaml#8329] [Q&A] [answered]
I am trying to activate a ProgressRing (ProgressRing.IsActive = true) prior to calling a synchronous methods in MailKit to send an email. With a good internet connection, the process usually takes about 3-4 seconds, during which time I'd like the ProgressRing to spin. But it seems that the MailKit threads take over and complete before the interface has time to react, and the ProgressRing just sits idle.

You can recreate this situation with the following simple code. How can I work around this?

XAML:

<ProgressRing x:Name="ProgressRingSendMail" IsActive="False""/>

C#:

    // start the progress bar
    ProgressRingSendMail.IsActive= true;
    Thread.Sleep(3000);
    ProgressRingSendMail.IsActive= false;

**Comment by kmgallahan** [comment, marked as answer]
Well, you shouldn't be hanging the UI thread as in most cases it is going to be completely frozen the whole time. Here though the GPU takes care of the animation once the UI has flushed the render cycle. The time to next flush would depend on your refresh rate, but ~20ms would handle most cases, so:

```cs
await Task.Delay(20);
```

after setting `IsActive`.

**Reply by greatoceansoftware** [reply]
Thanks. It works!
